drivers/backup/auth.go

```diff
@@ -264,8 +282,68 @@ func GetAllRoles() ([]KeycloakRoleRepresentation, error) {
 	return roles, nil
 }
 
+// GetRolesForUser lists all the available roles in keycloak for the provided username
+func GetRolesForUser(userName string) ([]KeycloakRoleRepresentation, error) {
+	fn := "GetRolesForUser"
+	headers, err := GetCommonHTTPHeaders(PxCentralAdminUser, PxCentralAdminPwd)
+	if err != nil {
+		logrus.Errorf("%s: %v", fn, err)
+		return nil, err
+	}
+	keycloakEndPoint, err := getKeycloakEndPoint(true)
+	if err != nil {
+		return nil, err
+	}
+	userID, err := FetchIDOfUser(userName)
+	if err != nil {
+		return nil, err
+	}
+	reqURL := fmt.Sprintf("%s/users/%s/role-mappings/realm/composite", keycloakEndPoint, userID)
+	method := "GET"
+	response, err := processHTTPRequest(method, reqURL, headers, nil)
+	if err != nil {
+		logrus.Errorf("%s: %v", fn, err)
+		return nil, err
+	}
+	var roles []KeycloakRoleRepresentation
+	err = json.Unmarshal(response, &roles)
+	if err != nil {
+		logrus.Errorf("%s: %v", fn, err)
+		return nil, err
+	}
+	return roles, nil
+}
+
+// ValidateUserRole will validate if a given user has the provided PxBackupRole mapped to it
+func ValidateUserRole(userName string, role PxBackupRole) bool {
+	roleMapping, err := GetRolesForUser(userName)
+	if err != nil {
+		return false
```

Comment:
retruning just false we will miss to log the error, fail it here logging the error or return the error

Reply:
Now returning bool and err. Then handling error and failing it.